tsdb/index/tsi1/partition.go

```diff
@@ -479,17 +522,19 @@ func (p *Partition) prependActiveLogFile() error {
 	}
 	p.activeLogFile = f
 
-	// Prepend and generate new fileset.
-	p.fileSet = p.fileSet.PrependLogFile(f)
+	// Prepend and generate new fileset but do not yet update the partition
+	newFileSet := p.fileSet.PrependLogFile(f)
 
 	// Write new manifest.
-	manifestSize, err := p.Manifest().Write()
+	manifestSize, err := p.manifest(newFileSet).Write()
 	if err != nil {
-		// TODO: Close index if write fails.
-		p.logger.Error("manifest write failed, index is potentially damaged", zap.Error(err))
+		p.logger.Error("manifest write failed, index is potentially damaged", zap.String("path", p.ManifestPath()), zap.Error(err))
+		p.closeOwner()
```

Comment:
Because we are closing the owning index, the new file in the manifest will be closed, whereas with this code, the new file will be ignored by the Index.Close().  However, if we do close the file, we don't leak resources, and it will be removed on next reboot because it's not in the Manifest file (because we failed to create it).